src/bootstrap/compile.rs

```diff
@@ -651,6 +651,12 @@ impl Step for Rustc {
             panic!("Cannot use and generate PGO profiles at the same time");
         }
 
+        // With LLD, we can use ICF (identical code folding) to reduce the executable size
+        // of librustc_driver/rustc and to improve i-cache utilization.
+        if builder.config.use_lld {
+            cargo.rustflag("-Clink-args=-Wl,--icf=all");
```

Comment:
This is specific to the compiler; it doesn't affect tools or the standard library. Is that intentionall?

Reply:
My main motivation was to try to reduce the size of `librustc_driver`, the largest dynamic library. But we can also try it for `libstd`.

Reply:
`icf=all` is a bit zealous, should be `safe`?

Reply:
In my experiments, safe didn't really do anything in terms of binary size. But we can try it.

Reply:
https://github.com/rust-lang/rust/pull/99455 let's see :)

Reply:
`icf=safe` doesn't improve binary sizes at all, sadly. There are still nice cycle wins, but these come mostly from using `lld`.